modules/storage/storage-account/main.bicep

```
@@ -86,75 +86,54 @@ type cors = {
   ]
 }
 
+@description('The blob service properties for blob soft delete.')
 type deleteRetentionPolicy = {
   allowPermanentDelete: bool
-  days: deleteRetentionPolicyDays?
+  @minValue(1)
+  @maxValue(365)
+  days: int?
   enabled: bool
 }
 
-@minValue(1)
-@maxValue(365)
-type deleteRetentionPolicyDays = int
-
 @description('Toggle to enable or disable versioning for Blob service of the Storage Account. Used only if enableBlobService is set to true.')
-type isBlobVersioningEnabled = bool
+type isVersioningEnabled = bool
 
+@description('The blob service property to configure last access time based tracking policy.')
 type lastAccessTimeTrackingPolicy = {
   blobType: [
     'string'
   ]
   enable: bool
   name: 'AccessTimeTracking'
-  trackingGranularityInDays: trackingGranularityInDays?
+  trackingGranularityInDays: 1?
 }
 
-type trackingGranularityInDays = 1
-
+@description('The blob service property to configure last access time based tracking policy.')
 type restorePolicy = {
-  days: restorePolicyDays?
+  @minValue(1)
+  @maxValue(365)
+  days: int?
   enabled: bool
 }
 
-@minValue(1)
-@maxValue(365)
-type restorePolicyDays = int
-
 type blobServiceContainerProperties = {
-  defaultEncryptionScope: defaultEncryptionScope?
-  denyEncryptionScopeOverride: denyEncryptionScopeOverride?
-  immutableStorageWithVersioning: immutableStorageWithVersioning?
-  publicAccess: blobContainerPublicAccess?
-}
-
-@description('Allowed values are \'Blob\', \'Container\' or \'None\'')
-type blobContainerPublicAccess = string
-
-type defaultEncryptionScope = string
-
-type denyEncryptionScopeOverride = string
-
-type immutableStorageWithVersioning = {
-  enabled: bool
+  defaultEncryptionScope: string?
+  denyEncryptionScopeOverride: string?
+  immutableStorageWithVersioning: {
+    enabled: bool
+  }?
+  @allowed['Blob', 'Container', 'None']
+  publicAccess: string?
```

```suggestion
  publicAccess: ('Blob' | 'Container' | 'None')?
```

`@allowed` is I believe only permitted on `param` statements.